Python/codegen.c

```diff
@@ -5046,8 +5060,11 @@ codegen_augassign(compiler *c, stmt_ty s)
         break;
     case Subscript_kind:
         VISIT(c, expr, e->v.Subscript.value);
-        if (is_two_element_slice(e->v.Subscript.slice)) {
-            RETURN_IF_ERROR(codegen_slice(c, e->v.Subscript.slice));
+        if (!is_constant_slice(e->v.Subscript.slice) &&
+            is_two_element_slice(e->v.Subscript.slice)) {
```

Comment:
We could change 

```
static bool
is_two_element_slice(expr_ty s)
{           
    return s->kind == Slice_kind &&
           s->v.Slice.step == NULL;
} 
```

to something like

```
static bool
apply_two_element_slice_optimization(expr_ty s)
{           
    return s->kind == Slice_kind &&
           s->v.Slice.step == NULL &&
           !is_constant_slice(s);
} 
```
and then the checks don't need to change to call both functions. Might be a bit easier to read.

Reply:
Good idea.